activejob/lib/active_job/arguments.rb

```diff
@@ -64,7 +64,7 @@ def deserialize(arguments)
         RUBY2_KEYWORDS_KEY, RUBY2_KEYWORDS_KEY.to_sym,
         OBJECT_SERIALIZER_KEY, OBJECT_SERIALIZER_KEY.to_sym,
         WITH_INDIFFERENT_ACCESS_KEY, WITH_INDIFFERENT_ACCESS_KEY.to_sym,
-      ]
+      ].to_set
```

Comment:
Also wondering if we're not better to just check for `start_with?("_aj_")`. It's both cheap and ensure that namespace is free for future keys.